table/column.go

```diff
@@ -371,7 +373,35 @@ func GetColOriginDefaultValue(ctx sessionctx.Context, col *model.ColumnInfo) (ty
 
 // GetColDefaultValue gets default value of the column.
 func GetColDefaultValue(ctx sessionctx.Context, col *model.ColumnInfo) (types.Datum, error) {
-	return getColDefaultValue(ctx, col, col.GetDefaultValue())
+	defaultValue := col.GetDefaultValue()
+	if !col.DefaultIsExpr {
+		return getColDefaultValue(ctx, col, defaultValue)
+	}
+	return getColDefaultExprValue(ctx, col, defaultValue)
+}
+
+func getColDefaultExprValue(ctx sessionctx.Context, col *model.ColumnInfo, defaultValue interface{}) (types.Datum, error) {
+	x, ok := defaultValue.(string)
+	if !ok {
+		// TODO: refine the error.
+		return types.Datum{}, errors.New("can't cast default value as expression string")
+	}
+	var defaultExpr ast.ExprNode
+	expr := fmt.Sprintf("select %s", x)
```

Comment:
Why we have to construct a statement and a new `parser` here? Can we simply allocate the ID from `nextval` just like the way for `AutoID`?

Reply:
Well, I saw your replies from @tangenta 's comments.

Reply:
this is mainly for parseing the expr string to an expression tree.
nextval should use 'eval()' to get the sequence value under the expression architecture.

generated column seems the same.